server/api/routers/product.ts

```diff
@@ -0,0 +1,63 @@
+import { z } from "zod"
+import { createTRPCRouter, publicProcedure } from "@/server/api/trpc";
+import ApiUploadProduct from "@/components/admin/product/productFunctions/apiUploadProducts";
+import ApiFetchProducts from "@/components/admin/product/productFunctions/apiFetchProducts";
+import ApiUpdateProduct from "@/components/admin/product/productFunctions/apiUpdateProduct";
+
+const ProductDataInterface = z.object({
+  pid: z.number(),
+  name: z.string(),
+  code: z.string(),
+  link: z.string(),
+  description: z.string(),
+  price: z.string(),
+  image: z.string() // Assuming image can be null or a string URL
+})
+const simplifiedProducts= z.object({
+  image: z.string(),
+  id:  z.string(),
+  pid: z.number(),
+  name: z.string(),
+  code:  z.string(),
+  price: z.string(),
+  link:  z.string(),
+  review: z.array(z.record(z.unknown())),
+  description:  z.string(),
+ })
```

Comment:
SOmetimes u r starting variables with capital letter and sometimes not, have to maintain consistency and start with small letter for variables.

for schemas so segregate it better arrange them something like below

`const schemas = {
  overview: z.object({
    totalTagAndCashCost: z.number().nullable(),
    totalVehicles: z.number().nullable(),
    totalTrips: z.number().nullable(),
    tollTrips: z.number().nullable(),
  }),

  totalTollCost: z.object({
    totalTagAndCashCost: z.number().nullable(),
  }),

  tollPaidDaily: z.object({
    totalTagAndCashCostByDate: z.array(
      z.object({
        date_val: z.string().nullable(),
        total_tag_and_cash_cost: z.coerce.number().nullable(),
      })
    ),
  }),

  tollPaidCountDaily: z.object({
    totalTollsPaidByDate: z.array(
      z.object({
        date_val: z.string().nullable(),
        total_tolls_paid: z.coerce.number().nullable(),
      })
    ),
  }),

  combinedTollFrequencyMap: z.object({
    combined_tolls: z.array(
      z.object({
        toll_loc_name_start: z.string(),
        entry_lat: z.number(),
        entry_lng: z.number(),
      })
    ),
  }),
};`